### ISSUE TITLE
How to use with 2-d model with 4-d input?

### ISSUE THREAD

**Issue by joshua-gould** [TheJacksonLaboratory/dask_relabeling#5]
The cellpose cyto3 model (https://github.com/MouseLand/cellpose/blob/main/notebooks/run_cyto3.ipynb) takes a 4-d array as input. How do I use your library with this model? Thanks

**Comment by fercer** [comment]
Hi @joshua-gould!

I took a look at the cellpose notebook and here are some recommendations on how to use `dask_relabeling` for that example.

1. You can define the segmentation function that is applied chunk-wise by wrapping the one shown in the original notebook as follows.
```
def cellposeLabel(im_chunk, model=None):
    masks_pred, flows, styles, diams = model.eval(im_chunk, diameter=0, channels=[1,2], niter=2000) # using more iterations for bacteria
    masks_pred = masks_pred.astype(np.int32)
    return masks_pred
```

2. Convert the images to Dask arrays.
```
import dask.array as da

im_da_list = [
    da.from_array(img, chunks=(2, 256, 256))
    for img in imgs
]
```

3. Use the relabeling tool as follows. Please notice that the `ndim` parameter has changes to `spatial_dims` for clarity (#6). So the "cyto3" model for 2D segmentation requires `spatial_dims=2` (X and Y axes). This doesn't count other axes, like the channel axes.
```
from dask_relabeling import relabel

masks_pred = [
    relabel.image2labels(
        img_da,
        seg_fn=cellposeLabel,
        overlaps=[32, 32],
        spatial_dims=2,
        segmentation_fn_kwargs={"model": model}
    )
    for img_da in im_da_list
]
```

5. Compute the predicted masks to continue with the notebook.
```
masks_pred = [masks.compute()
    for masks in masks_pred
]
```

The last point (4) is what you'd change for a dask.array.to_zarr or similar to store the labels without having to hold everything in RAM.

I executed these changes without problem in Colab. But please try these and let me know if it worked.

**Comment by joshua-gould** [comment]
Thanks. So you assume that the leading dimension are non-spatial dimensions, correct?

**Comment by fercer** [comment]
Yes, and this is for now hardcoded.

However, any permutation to the channels could be applied after converting the image into `dask.array` and before calling `relabel.image2labels`.

**Comment by joshua-gould** [comment]
Thanks!
